app/code/Magento/InventoryConfigurableProduct/Plugin/Block/Product/View/Type/AddOptionQtysPlugin.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryConfigurableProduct\Plugin\Block\Product\View\Type;
```

Comment:
the module where this logic should belong to should be `InventoryConfigurableProductFrontendUi`